ampersand-collection.js

```diff
@@ -276,16 +276,30 @@ extend(Collection.prototype, BackboneEvents, {
         }
     },
 
-    _deIndex: function (model) {
-        for (var name in this._indexes) {
-            delete this._indexes[name][model[name] || (model.get && model.get(name))];
+    _deIndex: function (model, attribute, value) {
+        if (undefined !== attribute) {
+            if (undefined === this._indexes[attribute]) throw new Error('Given attribute is not an index');
+            delete this._indexes[attribute][value];
+            return;
+        }
+        // Not a specific attribute
+        for (attribute in this._indexes) {
+            delete this._indexes[attribute][model[attribute] || (model.get && model.get(attribute))];
         }
     },
 
-    _index: function (model) {
-        for (var name in this._indexes) {
-            var indexVal = model[name] || (model.get && model.get(name));
-            if (indexVal) this._indexes[name][indexVal] = model;
+    _index: function (model, attribute) {
+        var indexVal;
+        if (undefined !== attribute) {
```

Comment:
It's a better practice to put constant first in a comparison. Consider this comparison typo issue:
``` javascript
if (attribute = 'aValue')
```
``` javascript
if ('aValue' = attribute)
```
The second form is really easier to detect

But I can change it if it breaks project coding style